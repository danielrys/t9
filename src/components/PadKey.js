// @flow
import * as React from "react"
import { TouchableOpacity, Text, StyleSheet } from "react-native"

// theme
import { Colors, Metrics } from "../themes"

// types
import type { KeyConfig } from "../types"

const styles = StyleSheet.create({
  container: {
    width: Metrics.padKey.baseDimension,
    height: Metrics.padKey.baseDimension,
    borderWidth: Metrics.borders.widths.base,
    borderColor: Colors.border,
    marginBottom: Metrics.spacings.medium,
  },
  number: {
    fontSize: Metrics.texts.sizes.big,
    color: Colors.textPrimary,
  },
  letters: {
    fontSize: Metrics.texts.sizes.small,
    color: Colors.textPrimaryMuted,
  },
})

type PadKeyProps = {
  keyConfig: KeyConfig,
  onPress: (number | string) => void,
}

export default class PadKey extends React.PureComponent<PadKeyProps> {
  handlePress = () => {
    const {
      keyConfig: { value },
      onPress,
    } = this.props
    onPress(value)
  }

  render() {
    const {
      keyConfig: { value, subvalue },
    } = this.props
    return (
      <TouchableOpacity style={styles.container} onPress={this.handlePress}>
        <Text style={styles.number}>{value}</Text>
        <Text style={styles.letters}>{subvalue}</Text>
      </TouchableOpacity>
    )
  }
}
