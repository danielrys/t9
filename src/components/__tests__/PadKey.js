import React from "react"
import renderer from "react-test-renderer"

// components
import PadKey from "../PadKey"

describe("PadKey component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PadKey
          onPress={() => undefined}
          keyConfig={{ value: 1, subvalue: "abc" }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
