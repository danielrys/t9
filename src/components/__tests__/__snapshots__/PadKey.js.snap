// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PadKey component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgb(150,150,150)",
      "borderRadius": 80,
      "borderWidth": 1,
      "height": 80,
      "justifyContent": "center",
      "marginBottom": 15,
      "opacity": 1,
      "width": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(0,0,0)",
        "fontSize": 24,
      }
    }
  >
    1
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(50,50,50)",
        "fontSize": 12,
      }
    }
  >
    abc
  </Text>
</View>
`;
